### PR TITLE
Prefer the local Quick Start in docs

### PR DESCRIPTION
**Pull request by rossturk**
I have been studying Airflow's web analytics, and noticed that the Docker Compose Quick Start is a page that stands out a bit - people spend a lot of time here compared to other pages. So I started to investigate.

The language on that page is currently very strong, discouraging its use for almost all situations.  I had a few conversations with @potiuk and others about this page and heard an opinion that `airflow standalone` is a better procedure for those just getting started. The standalone Quick Start procedure has no such disclaimers, and is far simpler. After using that method for a few projects, I think I agree.

Having a strongly worded warning on the Docker Compose procedure may not be the best way to communicate that the standalone procedure is better. Instead, I think the Docker Compose procedure should be made into a How To, and the standalone procedure should take its place. This will clearly communicate the preference for the standalone procedure without making the Docker Compose unavailable to those who rely upon it.

This PR is intended to:
1. Restore `start/local.rst` to `start.rst`
2. Move `start/docker.rst` and associated assets to `howto/docker-compose/`
3. Update links throughout to point at these new locations
4. Create redirects to support these moved resources
5. Reduce warnings in the Docker doc, now that it's not so prominent
6. Misc copy updates in Docker doc, since I was in there :) 

Signed-off-by: Ross Turk <ross@rossturk.com>

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: README.md
Comment (kaxil) [inline README.md:120]: This will become a broken link until a version is released with these doc changes.

Mostly a note for committers

**Review by kaxil — APPROVED** [review]

File: README.md
Comment (potiuk) [reply README.md:120]: We can add sphinx redirections.

Comment (potiuk) [reply README.md:120]: see `redirects.txt`

Comment (potiuk) [reply README.md:120]: Ah I see  it is added :)

File: docs/apache-airflow/howto/docker-compose/index.rst
Comment (potiuk) [inline docs/apache-airflow/howto/docker-compose/index.rst]: I think we need to also mention one thing here that any extensions of the docker-compose require experience in docker-compose. One of the problems that the "do not use docker compose" message was trying to fend-off is avoiding the users to ask this question:

> We are using the OFFICIAL docker compose from your page and we added this and that to add this and that feature and it does not work. Tell us how to do it.

 The assumption people have when looking at any of the documentation we publish on "airflow.apache.org" is that this is an official way and they expect it to work and they  expect things will work for them and that they will get support - because we officially publish the docker-compose. 

So far I have not seen anyone from the community (unlike in chart case) to step up and attempt to help such users when they raise the question "please help, this your official docker compose so surely you will help us when we have problems.

Are we ready for that ? I think not. Or at least I have not seen anyone even trying (this is why the disclaimers were there). In a way moving it out to HowTo makes the situation even worse, because it was "quick start" and we could tell "this is a quick start, don't expect more". 

I think we should addres it.

**Review by potiuk — CHANGES_REQUESTED** [review]
I like the direction. But I think we need a short, strong statement that we will be able to send people to when they make some modifications to docker compose. and expect support from the community, claiming this is "official chart". It does not have to be as strong or "push-backy" as the previous one, but I think poeople reading the how to need to bee aware about  clear expectations:

- they need to learn docker compose first 
- if they modify the docker compose, they are mostly on their own to debug any problems.

Unless of course there is someone in the community who volunteers to help people when they raise their issues.

File: docs/apache-airflow/howto/docker-compose/index.rst
Comment (potiuk) [reply docs/apache-airflow/howto/docker-compose/index.rst]: I do not know how to do it in a way that willl be short and smart, but I really think we need to to have a link where we send people to when they claim "this your official compose - we expect it to work, help us".

File: README.md
Comment (rossturk) [reply README.md:120]: Fortunately the current docs already have a redirect from `start.rst` to `start/index.rst` - this PR contains a redirect that reverses that.

I think / hope this will result in a smooth transition 👍

File: docs/apache-airflow/howto/docker-compose/index.rst
Comment (rossturk) [reply docs/apache-airflow/howto/docker-compose/index.rst]: I have added a more strongly-worded (but hopefully still friendly and empowering!) admonition and upgraded it from `warning` to `caution`. Let me know what you think!

Also, I added a paragraph right at the beginning that suggests users read the docker / docker-compose quick starts if they aren't familiar with them yet.

**Comment by rossturk** [comment]
> I like the direction. But I think we need a short, strong statement that we will be able to send people to when they make some modifications to docker compose. and expect support from the community, claiming this is "official chart". It does not have to be as strong or "push-backy" as the previous one, but I think poeople reading the how to need to bee aware about clear expectations:
> 
> * they need to learn docker compose first
> * if they modify the docker compose, they are mostly on their own to debug any problems.
> 
> Unless of course there is someone in the community who volunteers to help people when they raise their issues.

I completely understand your concerns 💯 Hopefully my changes address them - let me know if not.

**Comment by potiuk** [comment]
Did you push the changes:) ? I can't see the latest ones with caution :D .

Also I think maybe @mik-laj - ther original author of the Docker quickstart will have some comments ?

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
This is perfect!

**Comment by potiuk** [comment]
@mik-laj - any comments before we merge ?

**Review by mik-laj — APPROVED** [review]

**Comment by potiuk** [comment]
:tada:

**Comment by merobi-hub** [comment]
adding #protm for testing

**Comment by merobi-hub** [comment]
tagging with #PROTM for testing

**Comment by merobi-hub** [comment]
tagging with #protm for testing

**Pull request merged**
